pkg/ccl/backupccl/restoration_data.go

```diff
@@ -105,6 +124,28 @@ func (b *restorationDataBase) getSpans() []roachpb.Span {
 	return b.spans
 }
 
+// coveredByLaterReintroduction implements restorationData.
+func (b *restorationDataBase) coveredByLaterReintroduction(
+	spanStartKey roachpb.Key, backupEndTime hlc.Timestamp,
+) (bool, error) {
+	if b.backupCodec.ForSystemTenant() && bytes.HasPrefix(spanStartKey, keys.TenantPrefix) {
+		// During a restore of a tenant, the tenant's tables cannot get decoded, as
+		// restore spans are processed on the tenant level instead of the table or
+		// index level.
+		return false, nil
+	}
+	_, tablePrefix, err := b.backupCodec.DecodeTablePrefix(spanStartKey)
+	if err != nil {
+		return false, err
+	}
+	latestIntro, ok := b.latestEndTimesForReIntroducedTables[descpb.ID(tablePrefix)]
+	if !ok {
+		// span was never introduced
```

Comment:
nit: span was never reintroduced, given that introduced means that it was new span in between backups.